src/utils/process.rs

```diff
@@ -498,18 +514,29 @@ where
 
     */
 
-    let pid_range = my_pid.saturating_sub(10)..my_pid.saturating_add(20);
+    // Also `add` because `A_has_pid101 | delta_has_pid102`, but if A is a wrapper which then calls
+    // git (no `exec`), then the final pid of the git process might be 103 or greater.
+    let pid_range = my_pid.saturating_sub(10)..my_pid.saturating_add(10);
     for p in pid_range {
         // Processes which were not refreshed do not exist for sysinfo, so by selectively
         // letting it know about processes the `find_sibling..` function will only
         // consider these.
-        info.refresh_process(p);
+        if info.process(p).is_none() {
+            info.refresh_process(p);
+        }
     }
 
     match info.find_sibling_in_refreshed_processes(my_pid, &extract_args) {
         None => {
-            info.refresh_processes();
-            info.find_sibling_in_refreshed_processes(my_pid, &extract_args)
+            // The full scan is expensive and rarely successful, so disable it by default
+            if std::env::vars()
+                .any(|(k, v)| k == "DELTA_CALLING_PROCESS_QUERY_ALL" && v != "0" && v != "false")
```

Comment:
Better to use [`std::env::var`](https://doc.rust-lang.org/std/env/fn.var.html) here?

Reply:
I just never bothered to look up the equivalent for the `var()` call, which is `.map_or(false, ...)`, until now :)

And I think the underlying `getenv()` has to walk the environ list just the same.